runtime/lua/vim/treesitter/dev.lua

```diff
@@ -595,9 +600,14 @@ function M.edit_query(lang)
   end
   vim.cmd(cmd)
 
-  local parser = vim.treesitter._get_parser(buf, lang)
+  local parser = vim.treesitter.get_parser(buf, lang, { error = false })
   if not parser then
-    return nil, 'No parser available for the given buffer'
+    return nil,
+      string.format(
+        'Failed to show query editor for buffer: %s. No parser for lang: %s.',
```

Comment:
```suggestion
        'Failed to show query editor for buffer %s: no parser for lang %s.',
```

Nit, while something still has to be fixed anyway. (And similarly in other places; colons are not needed here.)